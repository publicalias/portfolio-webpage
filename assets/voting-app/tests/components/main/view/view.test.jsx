"use strict";

//local imports

const View = require("../../../../scripts/components/main/view/view");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { newPoll } = require("schemas/voting-app");
const { testMock } = require("test-helpers/meta-tests");
const { reactTests } = require("test-helpers/react-tests");

//setup

beforeAll(reactTests.setup);

//view

describe("view", () => {

  const { testMount, testShallow } = testWrapper(View);

  it("should match snapshot (default)", () => {

    const { wrapper } = testShallow();

    expect(wrapper).toMatchSnapshot();

  });

  it("should match snapshot (id)", () => {

    const { wrapper } = testShallow({ polls: [newPoll({ id: "id-a" })] }, {}, { location: { search: "?id=id-a" } });

    expect(wrapper).toMatchSnapshot();

  });

  it("should call viewClearState on load (default)", () => {

    const { props, wrapper } = testMount();

    const { actions: { viewClearState } } = props;

    wrapper.mount();

    testMock(viewClearState, []);

    wrapper.unmount();

  });

  it("should call viewClearState and metaGetPolls on load (id)", () => {

    const { props, wrapper } = testMount({}, {}, { location: { search: "?id=id-a" } });

    const { actions: { metaGetPolls, viewClearState } } = props;

    wrapper.mount();

    testMock(viewClearState, []);
    testMock(metaGetPolls, [null, "id-a"]);

    wrapper.unmount();

  });

});
