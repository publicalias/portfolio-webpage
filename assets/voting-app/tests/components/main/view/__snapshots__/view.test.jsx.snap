// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view should match snapshot (default) 1`] = `
<div
  className="c-ui__view"
>
  <ViewMenu
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleConfirm": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleConfirm": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "confirm": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": 0,
        "polls": Array [],
        "user": Object {},
        "view": Object {
          "add": "",
          "confirm": false,
          "settings": false,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    local={
      Object {
        "poll": Object {
          "author": "",
          "date": 0,
          "id": "",
          "options": Array [],
          "secret": false,
          "title": "",
          "users": Object {
            "created": "",
            "flagged": Array [],
            "hidden": Array [],
            "voted": 0,
          },
        },
      }
    }
    location={Object {}}
    match={Object {}}
    staticContext={Object {}}
  />
  <div
    className="c-poll-display"
  />
  <div
    className="c-poll-options"
  />
</div>
`;

exports[`view should match snapshot (id) 1`] = `
<div
  className="c-ui__view"
>
  <ViewMenu
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleConfirm": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleConfirm": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "confirm": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": 0,
        "polls": Array [
          Object {
            "author": "",
            "date": 0,
            "id": "id-a",
            "options": Array [],
            "secret": false,
            "title": "",
            "users": Object {
              "created": "",
              "flagged": Array [],
              "hidden": Array [],
              "voted": 0,
            },
          },
        ],
        "user": Object {},
        "view": Object {
          "add": "",
          "confirm": false,
          "settings": false,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    local={
      Object {
        "poll": Object {
          "author": "",
          "date": 0,
          "id": "id-a",
          "options": Array [],
          "secret": false,
          "title": "",
          "users": Object {
            "created": "",
            "flagged": Array [],
            "hidden": Array [],
            "voted": 0,
          },
        },
      }
    }
    location={
      Object {
        "search": "?id=id-a",
      }
    }
    match={Object {}}
    staticContext={Object {}}
  />
  <div
    className="c-poll-display"
  />
  <div
    className="c-poll-options"
  />
</div>
`;
